src/hooks/useCourtContracts.js

```diff
@@ -174,11 +197,110 @@ export function useDisputeActions() {
     [disputeManagerContract]
   )
 
-  return { draft, commit, reveal, leak, appeal, confirmAppeal }
+  const executeRuling = useCallback(
+    disputeId => {
+      return aragonCourtContract.executeRuling(disputeId, {
+        gasLimit: GAS_LIMIT,
+      })
+    },
+    [aragonCourtContract]
+  )
+  return {
+    approveFeeDeposit,
+    draft,
+    commit,
+    reveal,
+    leak,
+    appeal,
+    confirmAppeal,
+    executeRuling,
+  }
+}
+
+/**
+ *
+ * @param {string} disputeId id of the dispute
+ * @param {string} roundId id of the round
+ * @returns {Object} appeal deposit and confirm appeal deposit amounts
+ */
+export function useAppealDeposits(disputeId, roundId) {
+  const [appealDeposits, setAppealDeposits] = useState([bigNum(0), bigNum(0)])
+
+  const disputeManagerContract = useCourtContract(
+    CourtModuleType.DisputeManager,
+    disputeManagerAbi
+  )
+
+  useEffect(() => {
+    const getNextRoundDetails = async () => {
+      if (!disputeManagerContract) return
+      const nextRound = await disputeManagerContract.getNextRoundDetails(
+        disputeId,
+        roundId
```

Comment:
Probably something that would impact many places, but I’m wondering if we should have some sort of a retry mechanism for these calls, in case they fail for any reason.

Reply:
Yes, totally makes sense, created a `retryMax` function with a default number of attempts of 5. Let me know if you see any issues with the implementation

Reply:
LGTM!